api/app/serializers/annotation_serializer.rb

```diff
@@ -5,8 +5,9 @@ class AnnotationSerializer < ActiveModel::Serializer
 
   has_one :creator
   attributes :created_at, :end_char, :end_node, :id, :start_char, :start_node,
-             :text_section_id, :updated_at, :format, :resource_id, :creator_id,
-             :body, :private, :subject, :current_user_is_creator, :can_update_object,
+             :text_section_id, :text_section_name, :updated_at, :format,
```

Comment:
This seems problematic. For every annotation that exists in a text, the API will need to do an additional query for the corresponding text section, so you're in a N+1 query situations. I assume that the components on the frontend have access to the text and its sections, and we could determine the title from the text_section ID in the client, rather than making the API do more work.

Reply:
To be clear, "this" in that first sentence refers to the addition of :text_section_name